### PR TITLE
remove non-reuseable bits of MasterServer

### PR DESCRIPTION
**Pull request by deads2k**
Scrub `master.go` again.  I think I'm pretty happy with this shape.  I may promote `InstallAPIs` since we're likely to want it downstream.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35795)

### CODE REVIEW

File: pkg/master/master.go [hunk 1]
```diff
@@ -272,34 +240,22 @@ func (m *Master) InstallGeneralEndpoints(c *Config) {
 
 }
 
-func (m *Master) InstallAPIs(c *Config, restOptionsGetter genericapiserver.RESTOptionsGetter) {
+// InstallAPIs will use the restStorageProviders if they are enabled.
```
Comment (sttts) [inline pkg/master/master.go]: s/use/install the APIs for/

File: pkg/master/master.go [hunk 2]
```diff
@@ -272,34 +240,22 @@ func (m *Master) InstallGeneralEndpoints(c *Config) {
...
+func (m *Master) InstallAPIs(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter genericapiserver.RESTOptionsGetter, restStorageProviders ...genericapiserver.RESTStorageProvider) {
 	apiGroupsInfo := []genericapiserver.APIGroupInfo{}
 
-	// Install third party resource support if requested
-	// TODO seems like this bit ought to be unconditional and the REST API is controlled by the config
-	if c.GenericConfig.APIResourceConfigSource.ResourceEnabled(extensionsapiv1beta1.SchemeGroupVersion.WithResource("thirdpartyresources")) {
-		var err error
-		// TODO figure out why this isn't a loopback client
-		m.thirdPartyResourceServer.ThirdPartyStorageConfig, err = c.StorageFactory.NewConfig(extensions.Resource("thirdpartyresources"))
```
Comment (sttts) [inline pkg/master/master.go]: good that this is gone :+1:

File: pkg/registry/core/rest/storage_core.go [hunk 3]
```diff
@@ -239,3 +241,49 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generi
...
+
+func (p LegacyRESTStorageProvider) GroupName() string {
+	return api.GroupName
+}
+
+type componentStatusStorage struct {
+	storageFactory genericapiserver.StorageFactory
+}
+
+func (s componentStatusStorage) getServersToValidate() map[string]apiserver.Server {
```
Comment (sttts) [inline pkg/registry/core/rest/storage_core.go]: s/get//

File: pkg/registry/rbac/rest/storage_rbac.go [hunk 4]
```diff
@@ -52,7 +52,7 @@ type RESTStorageProvider struct {
 var _ genericapiserver.RESTStorageProvider = &RESTStorageProvider{}
```
Comment (sttts) [inline pkg/registry/rbac/rest/storage_rbac.go]: Do you mean

```golang
var _ genericapiserver.RESTStorageProvider = RESTStorageProvider{}
```

here?

**Comment by sttts** [comment]
Some nits.

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #35898

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #35898

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #35898

**Pull request merged**